primefaces/src/main/java/org/primefaces/component/export/TableExporter.java

```diff
@@ -95,33 +96,35 @@ else if (LangUtils.isNotBlank(column.getField())) {
 
         List<UIColumn> allColumns = table.getColumns();
         List<UIColumn> exportcolumns = new ArrayList<>(allColumns.size());
-        Map<String, ColumnMeta> columnMetadata = table.getColumnMeta();
         boolean visibleColumnsOnly = getExportConfiguration().isVisibleOnly();
+        AtomicBoolean hasNonDefaultSortPriorities = new AtomicBoolean(false);
+        final List<ColumnMeta> visibleColumnMetadata = new ArrayList<>(allColumns.size());
 
-        if (columnMetadata == null || columnMetadata.isEmpty()) {
-            table.forEachColumn(true, true, true, col -> {
-                if (col.isExportable() && (!visibleColumnsOnly || (visibleColumnsOnly && col.isVisible()))) {
-                    exportcolumns.add(col);
+        table.forEachColumn(true, true, true, column -> {
+            if (column.isExportable() && (!visibleColumnsOnly || (visibleColumnsOnly && column.isVisible()))) {
+                String metaColumnKey = column.getColumnKey();
```

Comment:
I am testing your PR locally and its telling me the above `metaColumnKey` variable is unused and can be removed.

Reply:
Sorry couldn't get the project loaded into my IDE. Just using a text editor doesn't highlight such oversights. I will correct it.